### PR TITLE
Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by Ouail-01**
> ### This pull request includes the following changes:

- [X] Add Person class with necessary instance variables and methods.
- [X] Add Student class that inherits from Person with additional methods and functionalities specific to students.
- [X] Add Teacher class that inherits from Person with additional methods and functionalities specific to teachers.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(age, classroom, parent_permission: true, name: 'Unknown')
+    super(age, parent_permission: parent_permission, name: name)
```
Comment (TedLivist) [inline student.rb]: - [x] Your implementation will return errors. The parameter `name`, as defined in the Person class, uses the normal equal sign for assignment, so it makes to still follow the convention. And `name` should come before `parent_permission`.

  ```
  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
  ```

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(age, specialization, parent_permission: true, name: 'Unknown')
+    super(age, parent_permission: parent_permission, name: name)
```
Comment (TedLivist) [inline teacher.rb]: - [x] Your implementation will return errors. The parameter `name`, as defined in the Person class, uses the normal equal sign for assignment, so it makes to still follow the convention. And `name` should come before `parent_permission`.
  
  ```
  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Ouail-01,

Good job so far!
✔️ Classes are different files
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: teacher.rb (hunk 2 above)
Comment (Ouail-01) [reply teacher.rb]: Thank you @TedLivist, for taking the time to provide your valuable review.

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(age, classroom, name: 'Unknown', parent_permission: true)
```
Comment (Ismailco) [inline student.rb]: - [x] As the previous reviewer mentioned kindly make sure to use `=` for the name parameter 
```suggestion
  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
```

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(age, specialization, name: 'Unknown', parent_permission: true)
```
Comment (Ismailco) [inline teacher.rb]: - [x] As the previous reviewer mentioned kindly make sure to use `=` for the name parameter 
```suggestion
  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
```

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Ouail-01, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Ouail-01,

✔️ Implementation of inheritance

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**